src/composite/MegaChart.js

```diff
@@ -115,6 +117,81 @@
         this._csvButton.click = function (a) {
             context.downloadCSV();
         };
+        
+        this._maximizeButton = new Button()
+            .classed({ "composite_MegaChart-Maximize": true })
+            .id(this.id() + "_maximize")
+            .value("")
+        ;
+        this._maximizeButton.click = function (a) {
+            var target = context.target();
+            var node = target;
+            
+            var isMaximized = target.classList.contains('__hpccisMaximized');
+            
+            //Find the layout_Grid ancestor
+            while(!node.classList.contains("layout_Grid") && node.parentElement){
+                node = node.parentElement;
+            }
```

Comment:
This could be useful as a general method in Widget:
```
    Widget.prototype.locateAncestor = function (classID) {
        var widget = this.locateParentWidget(this._target);
        while (widget) {
            if (widget.classID() === classID) {
                return widget;
            }
            widget = this.locateParentWidget(widget._target.parentNode);
        }
        return null;
    };
```